### PR TITLE
[BACKEND] B-delete-product-from-list

### LINKED ISSUE #408: [BACKEND] Implement Delete Product from List endpoint
This endpoint satisfies delete product item from list. The product and list ids should be given in the url.

### PR DESCRIPTION
**Pull request by bekir96**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Description

This PR includes delete product from list endpoint. Response body is:
image

endpoint url is `/lists/{id}/product/{product_id}`. `id` field represents product list and `product_id` represents id of product to be deleted.

# Test results

image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: ecommerce/backend/API/views/product_list.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        product_list_item = ProductListItem(product_list_id=product_list.first().pk, product_id=int(product_id))
+        product_list_item.save()
+
+        return Response({'status': { 'successful': True, 'message': "Product is successfully added to list."}})
+
+    elif request.method == "DELETE":
+        product_list = ProductList.objects.filter(id=int(list_id))
+        if len(product_list) == 0:
+            return Response({'status': { 'successful': False, 'message': "This product list is invalid."}})
```
Comment (omerfdeniz) [inline ecommerce/backend/API/views/product_list.py]:
```suggestion
            return Response({'status': { 'successful': False, 'message': "No such list exists."}})
```

File: ecommerce/backend/API/views/product_list.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        product_list.delete()
+
+    return Response({'status': { 'successful': True, 'message': "This product list is successfully deleted."}})
+
+
+@api_view(['POST', 'DELETE'])
+@permission_classes([permissions.AllowAnonymous])
+def manage_product_list_item(request, list_id, product_id):
+    jwt = authentication.JWTAuthentication()
+    user = jwt.authenticate(request=request)[0]
```
Comment (omerfdeniz) [inline ecommerce/backend/API/views/product_list.py]: You can also use the authenticated user in the request. You can reach it as
```suggestion
    user = request.user
```

**Review by omerfdeniz — CHANGES_REQUESTED** [review]
Well done, please check the comments before merge

File: ecommerce/backend/API/views/product_list.py (hunk 2 above)
Comment (bekir96) [reply ecommerce/backend/API/views/product_list.py]: jwt.authenticate() provide us if user is not authorized, this function return error message.

**Review by omerfdeniz — APPROVED** [review]
Well done, ready to merge

**Review by umutoksuz — APPROVED** [review]

**Pull request merged**